### DISCUSSION TITLE
Why was Rust chosen for Neovide rather than another stack like Electron?

### DISCUSSION THREAD

**Discussion by MoltenMonster** [neovide/neovide#2893] [Q&A] [answered]
Neovide is an excellent GUI for Nvim with its sharp text in visual effects, but I think if it was written in a web framework like Electron then it would open the door to lots of customization with CSS.

**Comment by Kethku** [comment, marked as answer]
Two main reasons:
- Performance. Having a web rendering engine behind the scenes takes a lot of control away from us to reduce resource usage and speed up the rendering process. People are frustrated with electron apps because of the (often accurate) perception that they use too many resources for what they give. We don't need an entire web browser renderer to build something like Neovide, so we owe it to our users to reach for better.
- Rust is fun to work with. Especially when I first started the project, I was pretty sick and tired of using javascript/typescript for everything. This project originally was an excuse to really learn Rust, and it worked great for that. Rust is a fun language, there was a gap in neovim front ends written in Rust, and Neovide fills that gap nicely. People like to contribute to projects written in Rust. Neovide wouldn't be where it is now without contributors and I see the decision to use Rust as a big part of that culture.
